src/conversion.jl

```diff
@@ -62,25 +64,29 @@ SparseArrays.sparse(A::WrappedMap) = sparse(A.lmap)
 Base.convert(::Type{SparseMatrixCSC}, A::WrappedMap) = convert(SparseMatrixCSC, A.lmap)
 
 # TransposeMap & AdjointMap
-for (TT, T) in ((AdjointMap, adjoint), (TransposeMap, transpose))
-    @eval Base.convert(::Type{AbstractMatrix}, A::$TT) = $T(convert(AbstractMatrix, A.lmap))
-    @eval SparseArrays.sparse(A::$TT) = $T(convert(SparseMatrixCSC, A.lmap))
+for (T, t) in ((AdjointMap, adjoint), (TransposeMap, transpose))
+    @eval Base.convert(::Type{AbstractMatrix}, A::$T) = $t(convert(AbstractMatrix, A.lmap))
+    @eval SparseArrays.sparse(A::$T) = $t(convert(SparseMatrixCSC, A.lmap))
 end
 
 # LinearCombination
-function Base.Matrix{T}(ΣA::LinearCombination{<:Any,<:Tuple{Vararg{MatrixMap}}}) where {T}
+function Base.Matrix{T}(ΣA::LinearCombination{<:Any, <:Tuple{Vararg{MatrixMap}}}) where {T}
     maps = ΣA.maps
     mats = map(A->getfield(A, :lmap), maps)
     return Matrix{T}(sum(mats))
 end
-function SparseArrays.sparse(ΣA::LinearCombination{<:Any,<:Tuple{Vararg{MatrixMap}}})
+function SparseArrays.sparse(ΣA::LinearCombination{<:Any, <:Tuple{Vararg{MatrixMap}}})
     maps = ΣA.maps
     mats = map(A->getfield(A, :lmap), maps)
     return convert(SparseMatrixCSC, sum(mats))
 end
 
 # CompositeMap
-function Base.Matrix{T}(AB::CompositeMap{<:Any,<:Tuple{MatrixMap,LinearMap}}) where {T}
+# Are all these methods below really useful?
+# The conversion to matrices is not something that belongs to ordinary use of LinearMaps.
+# It's rather for debugging purposes, where efficiency is not the primary concern.
+# Also, I don't think they are really that much more efficient
```

Comment:
Generally, the `Matrix` conversion is used in the Kronecker code in the vec trick, which is why I started introducing these methods (at least with the intention to write code that is faster than the columnwise mat-vec-muls):

https://github.com/Jutho/LinearMaps.jl/blob/b38c1ac1900d4751237bbef62927477017f4f7d0/src/kronecker.jl#L121-L131

`X` there is promoted to a `MatrixMap`, that's why we have all those `MatrixMap` conversion methods here.